tests/kuttl/test-suites/default/watcher-api/03-assert.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: watcher.openstack.org/v1beta1
+kind: WatcherAPI
+metadata:
+  finalizers:
+  - openstack.org/watcherapi
+  name: watcherapi-kuttl
```

Comment:
this is testign the self finaliser behvior.

if we replace this with creating the watcher API by using the top-level watcher cr we will also need to add the watcher controller finaliser here in the future.